### ISSUE TITLE
Feedback

### ISSUE THREAD

**Issue by katranci** [jiminny/jiminny-waveform#4]
https://github.com/jiminny/jiminny-waveform/blob/c9264b94b39ea979d70ee45aecd6935e26ec0a92/src/Components/Waveform/Indicator.jsx#L101

Keeping `shouldIndicatorShow` state in Redux seems unnecessary because it is used in `<Indicator>` component only.

**Comment by notargetbg** [comment]
You are right, but what if some other component wanted to know about `<Indicator>` show state? Perhaps displaying a tip on the top side of the screen or something else like banner..

What is the correct approach here? First create a local state, then lift it up if we need to?

**Comment by katranci** [comment]
> What is the correct approach here? First create a local state, then lift it up if we need to?

Yes, this is definitely what I would do. Every now and then we come across this situation where something feels like a shared thing even though there is no requirement for it. I tend to stick with the 
current world in these circumstances rather than predicting the future. This also leads to better developer experience because there is less decision making involved.

**Comment by notargetbg** [comment]
This is so true...

> This also leads to better developer experience because there is less decision making involved.
